src/lib.rs

```diff
@@ -343,87 +363,92 @@ where
     type Item = NvmLogResult<T, F::Error>;
 
     fn next(&mut self) -> Option<Self::Item> {
-        let current_index = self.next_log_addr;
-        let remaining_bytes = self.nvm_log.flash.capacity() - current_index as usize;
-
-        let first_byte = &mut [0][..remaining_bytes.min(1)];
-        match self.nvm_log.flash.read(current_index, first_byte) {
-            Err(e) => Some(Err(NvmLogError::Flash(e))),
-            Ok(()) => {
-                match first_byte.get(0).copied() {
-                    None => {
-                        if current_index != 0 {
-                            self.next_log_addr = 0;
-                            self.next()
-                        } else {
-                            // no bytes could be read; iterator is empty
+        'outer: loop {
```

Comment:
recursive calls overflow the stack in debug mode, so we have to use an explicit loop